KEP-4049: Add StorageCapacityScoring feature gate documents (#49801)
kubernetes/website: content/en/docs/concepts/scheduling-eviction/scheduling-framework.md

[NormalizeScore](#normalize-scoring) phase, the scheduler will combine node
scores from all plugins according to the configured plugin weights.

#### Capacity scoring {#scoring-capacity}

{{< feature-state feature_gate_name="StorageCapacityScoring" >}}

Nodes can be scored based on their available storage capacity. This only applies
when not using static provisioning
([dynamic provisioning](/docs/concepts/storage/dynamic-provisioning/) only), and

> this feature is currently in alpha.

Reviewer: Early feedback: I think this detail would belong in https://kubernetes.io/docs/reference/scheduling/config/#scheduling-plugins

Author: Thank you. I will address it once the work on [the k/k PR](https://github.com/kubernetes/kubernetes/pull/128184) is complete.

Author: I will add to that document once the feature gate is enabled by default.

[NormalizeScore](#normalize-scoring) phase, the scheduler will combine node
scores from all plugins according to the configured plugin weights.

#### Capacity scoring {#scoring-capacity}

{{< feature-state feature_gate_name="StorageCapacityScoring" >}}

Nodes can be scored based on their available storage capacity. Originally
extended `VolumeCapacityPriority` (for static provisioning) to dynamic
provisioning. Now handles both functions since `VolumeCapacityPriority` was
removed as a feature gate. Extends kube-scheduler's VolumeBinding plugin and
uses Storage Capacity Tracking supported CSI plugins. Available for
CSI-managed persistent volumes, including local storage when supported by the

> CSI driver. This feature is currently in alpha.

Reviewer 2: Most of these are not complete sentences, i.e. no subjects in them.
Could you please fix?

Author: Thank you. I addressed it.

[NormalizeScore](#normalize-scoring) phase, the scheduler will combine node
scores from all plugins according to the configured plugin weights.

#### Capacity scoring {#scoring-capacity}

{{< feature-state feature_gate_name="StorageCapacityScoring" >}}

This feature adds a scoring method for pod scheduling with
[the topology-aware volume provisioning](/blog/2018/10/11/topology-aware-volume-provisioning-in-kubernetes/).

> This feature eases to schedule pods on nodes with either the most or least

Reviewer 2: ```suggestion
This feature enables pods to be scheduled to nodes with either the most or least
```

Author: I addressed it.

> available storage capacity. This feature originally extended the

Reviewer 2: ? "extended" what?

Author: Originally, `StorageCapacityScoring` was a feature that added dynamic provisioning support to `VolumeCapacityPriority`, which only supported static provisioning. However, because `VolumeCapacityPriority` was going to be removed, `StorageCapacityScoring` became the feature gate that includes both static and dynamic provisioning.

Would "added" be more appropriate than "extended" in this context?

Author: I have modified this sentence as follows.

> Originally, `StorageCapacityScoring` was a feature that added dynamic provisioning support to `VolumeCapacityPriority`, which only supported static provisioning. However, because `VolumeCapacityPriority` was going to be removed, `StorageCapacityScoring` became the feature gate that includes both static and dynamic provisioning.

Reviewer 2: How about this?
> The feature gate `VolumeCapacityPriority` was used in v1.32 to support storage that are
> statically provisioned. Starting from v1.33, the new feature gate `StorageCapacityScoring`
> replaces the old `VolumeCapacityPriority` gate with added support to dynamically provisioned storage.
> When `VolumeCapacityScoring` is enabled, the VolumeBinding plugin in the kube-scheduler is extended
> to score Nodes based on the storage capacity on each of them.
> This feature is applicable to all CSI volumes, including local storage backed by
> a CSI driver.

Author: Thank you very much!

> When `VolumeCapacityScoring` is enabled,

(I think this is just a typo) It should be `StorageCapacityScoring` now.

> This feature is applicable to all CSI volumes,

The phrase "all CSI volumes" could be a bit misleading. This feature requires storage capacity information obtained from [Storage Capacity](https://kubernetes.io/docs/concepts/storage/storage-capacity/). Therefore, CSI volumes that do not support [Storage Capacity](https://kubernetes.io/docs/concepts/storage/storage-capacity/) are not covered by this feature.

Nevertheless, thank you. I will try to revise the text based on your comments.

Author: I addressed it.
